### PR TITLE
allow for multiple _prob_param files

### PR DESCRIPTION
**Pull request by zingale**
we now search all of the Castro source directories
the problem's own parameters are always parsed last.
We also get rid of _default_prob_params in Source/problems and just treat it like
any other _prob_params file.

## PR checklist

- [ ] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

**Comment by zingale** [comment]
tests: http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2020-12-20-002/index.html
and then: http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2020-12-20-003/index.html

**Comment by zingale** [comment]
this also starts some cleaning up of the python script to use f-strings

File: Exec/Make.auto_source [hunk 1]
```diff
@@ -47,8 +47,12 @@ CEXE_headers += prob_parameters.H
 $(CASTRO_AUTO_SOURCE_DIR)/prob_parameters.cpp: $(CASTRO_AUTO_SOURCE_DIR)/prob_params_auto.F90
 
 PROB_PARAMS_TEMPLATE := $(TOP)/Util/scripts/prob_params.template
-PROB_PARAMETERS := $(PROBLEM_DIR)/_prob_params
-DEFAULT_PROB_PARAMETERS := $(TOP)/Source/problems/_default_prob_params
+# search Bdirs for _prob_params files
+PROB_SEARCH := $(foreach dir, $(Bdirs), $(TOP)/$(dir))
+PROB_PARAMETERS := $(shell $(TOP)/Util/scripts/findparams.py $(PROB_SEARCH))
```
Comment (maxpkatz) [inline Exec/Make.auto_source:52]: Can't you also do this in GNU make with a wildcard on $(Bdirs)/_prob_params, instead of writing a new script?

Comment (zingale) [reply Exec/Make.auto_source:52]: ah indeed, I think maybe that could work.  Let me try.

Comment (zingale) [reply Exec/Make.auto_source]: okay, that's implemented.  Sedov compiles okay with the changes

**Review by maxpkatz — APPROVED** [review]

**Pull request merged**